hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddMapConfigMessageTask.java

```diff
@@ -84,13 +81,11 @@ protected IdentifiedDataSerializable getConfig() {
         config.setMaxIdleSeconds(parameters.maxIdleSeconds);
         config.setMaxSizeConfig(new MaxSizeConfig(parameters.maxSizeConfigSize,
                 MaxSizeConfig.MaxSizePolicy.valueOf(parameters.maxSizeConfigMaxSizePolicy)));
-        Version clusterVersion = nodeEngine.getClusterService().getClusterVersion();
-        if (clusterVersion.isGreaterOrEqual(V3_10) && parameters.mergeBatchSizeExist) {
+        if (parameters.mergeBatchSizeExist) {
             MergePolicyConfig mergePolicyConfig = mergePolicyConfig(true, parameters.mergePolicy,
                     parameters.mergeBatchSize);
             config.setMergePolicyConfig(mergePolicyConfig);
         } else {
-            // RU_COMPAT_3_9
             config.setMergePolicy(parameters.mergePolicy);
```

Comment:
Not so sure about removing this line. It is triggered if `parameters.mergeBatchSizeExist==false`.